client/src/systems/network_entity_update.rs

```diff
@@ -69,12 +79,41 @@ impl<'s> System<'s> for NetworkEntityStateUpdateSystem {
                 EntityType::Player => SpriteId::Player,
             };
 
-            entities.build_entity()
-                .with(net_id, &mut network_ids)
-                .with(transform, &mut transforms)
-                .with(sprite_resource.sprite_render_for(sprite_id), &mut sprite_renders)
+            lazy.create_entity(&entities)
+                .with(net_id)
+                .with(transform)
+                .with(sprite_resource.sprite_render_for(sprite_id))
                 .build();
         }
+
+        let deaths = death_event_channel.read(&mut self.death_reader);
+        if let Err(error) = Self::handle_deaths(deaths, &lazy, &sprite_resource, &entities, time.absolute_time()) {
+            log::error!("Failed to handle player death event: {}", error);
+        }
+    }
+}
+
+impl<'s> NetworkEntityStateUpdateSystem {
+    fn handle_deaths<D: IntoIterator<Item=&'s PlayerDeath>>(
+        deaths: D,
+        lazy: &LazyUpdate,
+        sprite_resource: &SpriteResource,
+        entities: &Entities<'_>,
+        current_time: Duration,
+    ) -> anyhow::Result<()> {
+        deaths.into_iter().for_each(|death| {
+            let transform = {
+                let mut transform = Transform::default();
+                transform.set_translation_xyz(death.position.x, death.position.y, -0.9);
```

Comment:
On the modular rendering branch I have changed the camera position a bit, so z will mean meters and 0.0 is the ground level. I wrote (positive) `0.1` for the corpse.
I wanted to mention it, because it might not be an obvious conflict for git